### ISSUE TITLE
Leaving text tool cancels the screenshot

### ISSUE THREAD

**Issue by J4bbi** [flameshot-org/flameshot#3333]
### Flameshot Version

Flameshot v11.0.0 (11.0.0-2 Debian

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

ubuntu: 22.04

### Description

When editing a screenshot I can use any of the tools and switch between them with the exception of the text tool.

I can succcessfully add a text to the screenshot, but if I then switch to any other tool the screenshot is aborted.

### Steps to reproduce

1. run flameshot gui
2.  select a region of the monitor
3. edit the screenshot using any tool other than the text tool
4. add a text to the screenshot
5. select another tool

### Screenshots or screen recordings

I tried three screenrecording software. Peek, Kooha and ubuntu native screen recording. None of them caught the flameshot gui.

### System Information

16:34  $ flameshot --version                                                                                    16:34:11
Flameshot v11.0.0 (11.0.0-2 Debian)
Compiled with Qt 5.15.2

6:38  $ inxi --width 80 --system --graphics   
System:
  Host: jabbi-CREM-WXX9 Kernel: 6.2.0-32-generic x86_64 bits: 64
    Desktop: GNOME 42.9 Distro: Ubuntu 22.04.3 LTS (Jammy Jellyfish)
Graphics:
  Device-1: AMD Cezanne driver: amdgpu v: kernel
  Device-2: ShineTech HD Camera type: USB driver: uvcvideo
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: amdgpu,ati
    unloaded: fbdev,modesetting,radeon,vesa gpu: amdgpu
    resolution: 2520x1680~60Hz
  OpenGL: renderer: RENOIR (renoir LLVM 15.0.7 DRM 3.49 6.2.0-32-generic)
    v: 4.6 Mesa 23.0.4-0ubuntu1~22.04.1

$ loginctl show-session $(loginctl show-user $(whoami) -p Display --value) -p Type --value
x11

16:40  $ xrandr --listactivemonitors                                                                            16:40:15
Monitors: 1
 0: +*eDP 2520/338x1680/226+0+0  eDP

16:45  $ sudo update-pciids  # you might need to run this with sudo                                             16:45:09
         lspci | grep -i 'vga\|3d\|2d'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  291k  100  291k    0     0   364k      0 --:--:-- --:--:-- --:--:--  364k
Done.
03:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Cezanne [Radeon Vega Series / Radeon Vega Mobile Series] (rev c5)

16:46  $ sudo lshw -class display                                                                               16:46:54
  *-display                 
       description: VGA compatible controller
       product: Cezanne [Radeon Vega Series / Radeon Vega Mobile Series]
       vendor: Advanced Micro Devices, Inc. [AMD/ATI]
       physical id: 0
       bus info: pci@0000:03:00.0
       logical name: /dev/fb0
       version: c5
       width: 64 bits
       clock: 33MHz
       capabilities: pm pciexpress msi msix vga_controller bus_master cap_list fb
       configuration: depth=32 driver=amdgpu latency=0 mode=2520x1680 resolution=2520,1680 visual=truecolor xres=2520 yres=1680
       resources: iomemory:fc0-fbf iomemory:fc0-fbf irq:58 memory:fce0000000-fcefffffff memory:fcf0000000-fcf01fffff ioport:1000(size=256) memory:d0400000-d047ffff

16:47  $ uname -a                                                                                               16:47:02
Linux jabbi-CREM-WXX9 6.2.0-32-generic #32~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Aug 18 10:40:13 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

<details>
<summary>The dbus-monitor content</summary>

```

signal time=1694706505.024060 sender=org.freedesktop.DBus -> destination=:1.183 serial=2 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameAcquired
   string ":1.183"
signal time=1694706505.024077 sender=org.freedesktop.DBus -> destination=:1.183 serial=4 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
   string ":1.183"
method call time=1694706513.748411 sender=:1.185 -> destination=org.freedesktop.DBus serial=12 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=AddMatch
   string "type='signal',sender='org.freedesktop.DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0='org.a11y.Bus',arg1=''"
method call time=1694706513.748441 sender=:1.185 -> destination=org.freedesktop.DBus serial=13 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameHasOwner
   string "org.a11y.Bus"
method call time=1694706513.748554 sender=:1.185 -> destination=org.freedesktop.DBus serial=14 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=AddMatch
   string "type='signal',sender='org.freedesktop.DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0='org.a11y.Bus'"
method call time=1694706513.748581 sender=:1.185 -> destination=org.freedesktop.DBus serial=15 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=GetNameOwner
   string "org.a11y.Bus"
method call time=1694706513.748663 sender=:1.185 -> destination=org.a11y.Bus serial=16 path=/org/a11y/bus; interface=org.freedesktop.DBus.Properties; member=Get
   string "org.a11y.Status"
   string "ScreenReaderEnabled"
method call time=1694706513.748913 sender=:1.185 -> destination=org.a11y.Bus serial=17 path=/org/a11y/bus; interface=org.freedesktop.DBus.Properties; member=Get
   string "org.a11y.Status"
   string "IsEnabled"



```

</details>

**Comment by mmahmoudian** [comment]
This is duplicated. Please search in open and closed issues before creating a new one. It can save your time. This issue is already fixed. Also your Flameshot version is pretty old. I suggest updating to the lastest version to get new features and fixed bugs. More information here:

https://flameshot.org/docs/installation/installation-linux/

**Comment by J4bbi** [comment]
@mmahmoudian , I did search on issues, but clearly not well enough

I also tried version 12. Which I should have mentioned, and that did not seem to make any diference.

apt installs version 11 as the latest

```
13:15  $ sudo apt install flameshot                                                                             13:15:22
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
flameshot is already the newest version (11.0.0-2).
```

**Comment by mmahmoudian** [comment]
> I did search on issues, but clearly not well enough

Yes, probably you didn't use good search terms. I didn't remember by heart what was the first post about this and when it was solved. So used `is:issue is:closed text tool ` and from [the results](https://github.com/flameshot-org/flameshot/issues?q=is%3Aissue+is%3Aclosed+text+tool) I manages to fish these out:

- #3265
- #3232
- #3120
- #2860

You can see in those that this issue is fixed in #2369 

>  also tried version 12. Which I should have mentioned, and that did not seem to make any diference.

I don't know how you have installed it, but make sure to remove the v11.0.0 before installing v12.1.0,

> apt installs version 11 as the latest

Well, it is not `apt` fault. Your distro of choice (i.e Ubuntu) is at this point very behind. This is the nature of point-release distros (e.g., Ubuntu, Debian, OpenSuse Leap, Fedora). If you move to a rolling-release distro (e.g., Arch, Manjaro, OpenSuse Tumbleweed, Nix, Guix) you will generally have more updated versions. For more information about point-release vs. rolling-release, you can read this link: https://linuxiac.com/rolling-vs-point-release-linux-distributions/


There is also a very easy way to check which distro has which version of Flameshot:
https://repology.org/project/flameshot/versions

Now, in your case, I personally do not recommend you to change your distro just because of one screenshot tool (although it is the best of the best 😉 ). You can get the latest version using any of the dostro-agnostic solutions. We have explained it one our website:

https://flameshot.org/docs/installation/installation-linux/#distro-agnostic

**Comment by J4bbi** [comment]
Thanks for your detailed response, it's very helpful and thanks for your efforts in developing this great OS tool
